src/gfx_layout_icu.cpp

```diff
@@ -381,6 +381,30 @@ std::vector<ICURun> ItemizeStyle(std::vector<ICURun> &runs_current, FontMap &fon
 	return new ICUParagraphLayout(runs, buff, length);
 }
 
+/* static */ std::unique_ptr<icu::BreakIterator> ICUParagraphLayoutFactory::break_iterator;
+
+/**
+ * Initialize data needed for the ICU layouter.
+ */
+/* static */ void ICUParagraphLayoutFactory::InitializeLayouter()
+{
+	auto locale = icu::Locale(_current_language->isocode);
+	UErrorCode status = U_ZERO_ERROR;
+	ICUParagraphLayoutFactory::break_iterator.reset(icu::BreakIterator::createLineInstance(locale, status));
+	assert(U_SUCCESS(status));
+}
+
+/**
+ * Get a thread-safe line break iterator.
+ * @returns unique_ptr managed BreakIterator instance.
+ */
+/* static */ std::unique_ptr<icu::BreakIterator> ICUParagraphLayoutFactory::GetBreakIterator()
+{
+	assert(ICUParagraphLayoutFactory::break_iterator != nullptr);
+
+	return std::unique_ptr<icu::BreakIterator>(ICUParagraphLayoutFactory::break_iterator->clone());
```

Comment:
Why not a `shared_ptr` instead?

Reply:
Because there's no need for a shared_ptr. The main iterator instance is not thread-safe, using a `clone()` gets around that without having to recreate the iterator from scratch. unique_ptr is used so that we know it's deleted properly.

Alternatively, like the previous version, the main iterator can be thread_local, but then this method of initialization doesn't work.

(I'm not sure if the break iterator even needs to be thread-safe, but I think defaulting to assuming it should be is wise.)